bundles/cif-virtual-catalog/src/main/java/com/adobe/cq/commerce/virtual/catalog/data/impl/ProductBindingCreator.java

```diff
@@ -176,8 +177,15 @@ private void processAddition(ResourceChange change) {
         LOG.debug("Check if we already have a binding at {}", bindingPath);
 
         try {
-            LOG.debug("Creating a new resource at {}", parent.getPath() + "/" + bindingName);
+            LOG.debug("Creating a new resource at {}", bindingPath);
             ResourceUtil.getOrCreateResource(resolver, bindingPath, mappingProperties, "", true);
+
+            if (contentResource != null) {
+                LOG.debug("Adding {} property at {}", Constants.PN_CATALOG_PATH, contentResource.getPath());
+                ModifiableValueMap map = contentResource.adaptTo(ModifiableValueMap.class);
+                map.put(Constants.PN_CATALOG_PATH, bindingPath);
+                resolver.commit();
```

Comment:
This code runs asynchronously. We should be careful with saving changes to the original configuration because someone might modify it in the UI in the same time (e.g. I saved my changes, but then I realized something is wrong and I immediately open the properties page again and save, all in the same time in which this event handler runs).

One solution would be to call a `resourceResolver.refresh()` before getting the `ModifiableValueMap`.

Reply:
`resolver.refresh();` is already called at the top of the `processAddition` method ... so I'm not sure if we really want/need to add an extra call.